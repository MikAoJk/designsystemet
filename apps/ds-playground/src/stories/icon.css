.myIconClassname {
  fill: var(--palette-denim-100);
}
