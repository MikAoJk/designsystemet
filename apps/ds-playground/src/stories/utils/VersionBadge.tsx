import { ReactNode } from 'react';

interface VersionBadgeProps {
  packageName: string;
}

export const VersionBadge = ({ packageName }: VersionBadgeProps): ReactNode => {
  let color = 'green';

  let version = process.env[`@skatteetaten/ds-${packageName}`] ?? '..';
  if (process.env['NODE_ENV'] === 'development') {
    color = 'red';
    version = 'developing';
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    color = 'blue';
  }

  const normalizedPackageName = packageName.replaceAll('-', '--');
  const normalizedVersion = version.replaceAll('-', '--').replaceAll('_', '__');

  return (
    <img
      src={`https://img.shields.io/badge/%40skatteetaten%2Fds--${normalizedPackageName}-${normalizedVersion}-${color}?logo=npm`}
      alt={`@skatteetaten/ds-${packageName}@${version}`}
    />
  );
};
