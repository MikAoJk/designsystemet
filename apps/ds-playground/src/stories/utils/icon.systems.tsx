import { SystemSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const SystemSVGPaths = findSVGPathElements(SystemSVGPathsAndIcons);
export const SystemIconLabels = makeLabelsFromSVGPathElements(SystemSVGPaths);
