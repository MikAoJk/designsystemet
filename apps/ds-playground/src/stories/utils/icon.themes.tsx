import { ThemeSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const ThemeSVGPaths = findSVGPathElements(ThemeSVGPathsAndIcons);
export const ThemeIconLabels = makeLabelsFromSVGPathElements(ThemeSVGPaths);
