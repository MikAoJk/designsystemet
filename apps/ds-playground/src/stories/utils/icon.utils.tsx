import {
  SystemSVGPathsAndIcons,
  ThemeSVGPathsAndIcons,
} from '@skatteetaten/ds-icons';

export type SVGPathAndIconType =
  | typeof SystemSVGPathsAndIcons
  | typeof ThemeSVGPathsAndIcons;
export type SVGPathAndIconKey = keyof SVGPathAndIconType;

export const makeLabelsFromSVGPathElements = function (
  SVGPathElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, string> {
  const labels: Record<string, string> = {};
  Object.keys(SVGPathElements).forEach((name: string) => {
    labels[name] = name.replace('SVGpath', '');
  });

  return labels;
};

export const findSVGPathElements = function (
  SVGPathAndIconElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, SVGPathAndIconType[SVGPathAndIconKey]> {
  const elements: Record<string, SVGPathAndIconType[SVGPathAndIconKey]> = {};
  Object.keys(SVGPathAndIconElements).forEach((elementName: string) => {
    if (elementName.endsWith('SVGpath')) {
      elements[elementName] =
        SVGPathAndIconElements[elementName as SVGPathAndIconKey];
    }
  });

  return elements;
};
