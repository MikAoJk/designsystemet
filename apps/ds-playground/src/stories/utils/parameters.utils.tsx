import { ArgsTable, Stories, PRIMARY_STORY } from '@storybook/addon-docs';
import type { Parameters } from '@storybook/csf';

type DesignParameters = {
  name: string;
  url: string;
};

export const getParameters = function (
  designParameters: DesignParameters[]
): Parameters {
  return {
    docs: {
      page: (): JSX.Element => {
        return (
          <>
            <Stories title={''} />
            <ArgsTable story={PRIMARY_STORY} />
          </>
        );
      },
    },
    design: designParameters.map((designParameter: DesignParameters) => {
      return {
        name: designParameter.name,
        type: 'figma',
        url: designParameter.url,
      };
    }),
    backgrounds: {
      values: [
        { name: 'Svart', value: '#000' },
        { name: 'Hvit', value: '#fff' },
      ],
    },
  };
};
