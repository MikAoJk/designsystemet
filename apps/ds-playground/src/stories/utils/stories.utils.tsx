export const exampleParameters = {
  docs: {
    source: {
      type: 'code',
      language: 'tsx',
      transform: (source: string): string => {
        return source
          .replace('render: (_args): JSX.Element', 'const Examples = ()')
          .replace(
            'render: function Render(_args): JSX.Element ',
            'const Examples = () => '
          )
          .replace('satisfies Story', '');
      },
    },
  },
  /* _args i en story må finnes slik at props i fanen Controls skjules og er prefikset med underscore for å unngå eslint klaging */
  controls: {
    exclude: /.*/,
  },
};
