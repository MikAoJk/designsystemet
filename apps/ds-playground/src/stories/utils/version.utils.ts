export const getVersion = (
  packageName: string
): {
  major: string;
  minor: string;
  patch: string;
  postfix?: string;
  style?: Record<string, string>;
} => {
  const ver = process.env[`@skatteetaten/${packageName}`] ?? '..';
  const [major, minor, patch] = ver.split('.');
  const majorWithPackageName = packageName ? `${packageName}@${major}` : major;

  const style = {
    color: 'var(--semantic-page-foreground)',
    background: 'var(--semantic-success-background)',
    'font-weight': 'var(--font-weight-bold)',
    'font-size': '13px',
  };

  if (process.env['NODE_ENV'] === 'development') {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: 'develop',
      style: {
        ...style,
        background: 'var(--semantic-danger-background)',
      },
    };
  }
  if (process.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    return {
      major: majorWithPackageName,
      minor,
      patch,
      postfix: process.env['STORYBOOK_BUILD_VERSION'],
      style: {
        ...style,
        background: 'var(--semantic-warning-background)',
      },
    };
  }

  return {
    major: majorWithPackageName,
    minor,
    patch,

    style: {
      ...style,
    },
  };
};
