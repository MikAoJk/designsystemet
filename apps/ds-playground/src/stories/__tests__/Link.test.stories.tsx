import { useState } from 'react';

import { Link, LinkProps } from '@skatteetaten/ds-buttons';
import { linkColorArr } from '@skatteetaten/ds-core-utils';
import { AddOutlineSVGpath, CalendarSVGpath } from '@skatteetaten/ds-icons';
import { expect } from '@storybook/jest';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { userEvent, waitFor, within } from '@storybook/testing-library';

import { wrapper } from './testUtils/storybook.testing.utils';
import { SystemSVGPaths } from '../utils/icon.systems';

const elementId = 'htmlId';
const systemIconViewBox = '0 0 24 24';
const defaultLinkText = 'Er du pendler?';

const verifyAttribute =
  (attribute: string, expectedValue: string) =>
  async ({ canvasElement }: { canvasElement: HTMLElement }): Promise<void> => {
    const canvas = within(canvasElement);
    await expect(canvas.getByRole('link')).toHaveAttribute(
      attribute,
      expectedValue
    );
  };
export default {
  component: Link,
  title: 'Tester/Link',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    isExternal: { table: { disable: true } },
    color: { table: { disable: true } },
    svgPath: {
      table: { disable: true },
      options: Object.keys(SystemSVGPaths),
      mapping: SystemSVGPaths,
    },
    title: { table: { disable: true } },
    // HTML
    href: { table: { disable: true } },
    target: { table: { disable: true } },
    // Aria
    ariaDescribedby: { table: { disable: true } },
    // Events
    onBlur: { table: { disable: true } },
    onClick: { table: { disable: true } },
    onFocus: { table: { disable: true } },
  },
} as ComponentMeta<typeof Link>;

const Template: ComponentStory<typeof Link> = (args) => (
  <div className={'noTransition'} data-test-block>
    <Link {...args} onClick={(e): void => e.preventDefault()}>
      {/* eslint-disable-next-line testing-library/no-node-access */}
      {args.children}
    </Link>
  </div>
);

const defaultArgs: LinkProps = {
  href: '#root',
  children: defaultLinkText,
};

// Når Link har en ref, så får dom a elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';
WithRef.args = {
  ...defaultArgs,
  ref: (instance: HTMLAnchorElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};
WithRef.argTypes = {
  ...WithRef.argTypes,
  ref: { table: { disable: false } },
};
WithRef.parameters = {
  imageSnapshot: { disable: true },
};
WithRef.play = verifyAttribute('id', 'dummyIdForwardedFromRef');

// Når Link har en id, så har a-element id
// Når Link har en custom CSS, så vises custom stil
// Når Link har en lang, så har a-element lang
// Når Link har dataTestid, så har a-elementet data-testid satt
export const WithAttributes = Template.bind({});
WithAttributes.storyName = 'With Attributes(FA2-5)';
WithAttributes.args = {
  ...defaultArgs,
  id: elementId,
  className: 'dummyClassname',
  lang: 'nb',
  'data-testid': '123ID',
};
WithAttributes.argTypes = {
  ...WithAttributes.argTypes,
  id: { table: { disable: false } },
  className: { table: { disable: false } },
  lang: { table: { disable: false } },
  'data-testid': { table: { disable: false } },
};
WithAttributes.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const link = canvas.getByRole('link');
  await expect(link).toHaveClass('dummyClassname');
  await expect(link).toHaveAttribute('id', elementId);
  await expect(link).toHaveAttribute('lang', 'nb');
  await expect(link).toHaveAttribute('data-testid', '123ID');
};

// Når Link instansieres, så får den riktige default-verdier og rendrer riktig i ulike tilstander
export const Defaults = Template.bind({});
Defaults.storyName = 'Defaults (A1 delvis, A2, A3 delvis, B1)';
Defaults.args = {
  ...defaultArgs,
};
Defaults.argTypes = {
  ...Defaults.argTypes,
  href: { table: { disable: false } },
  children: { table: { disable: false } },
};
Defaults.parameters = {
  imageSnapshot: {
    hover: `${wrapper} > a`,
    focus: `${wrapper} > a`,
    click: `${wrapper} > a`,
  },
};
Defaults.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const link = canvas.getByRole('link');
  await expect(link).not.toHaveAttribute('rel');
  await expect(link).not.toHaveAttribute('target');

  // eslint-disable-next-line testing-library/no-node-access
  const svg = link.querySelector('svg');
  await expect(svg).not.toBeInTheDocument();
  expect(link).toHaveTextContent(defaultLinkText);
};

// Når Link har en veldig lang tekst så skal tekst venstrejusteres
export const WithLongText = Template.bind({});
WithLongText.storyName = 'With Long Text (A1 delvis)';
WithLongText.args = {
  ...defaultArgs,
  children:
    'Denne lenken har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift med tekst som alltid er venstrejustert uansett om ikon eller ikke.',
};
WithLongText.argTypes = {
  ...WithLongText.argTypes,
  children: { table: { disable: false } },
};

// Når Link har en veldig lang tekst uten breaking space så skal det brekke over flere linjer
export const WithLongTextBreaking = Template.bind({});
WithLongTextBreaking.storyName = 'With Long Text And Breaking (A1 delvis)';
WithLongTextBreaking.args = {
  ...defaultArgs,
  children:
    'Dennelenkenharenveldiglangtekst.Sålangatdentvingerframlinjeskiftmedtekstsomalltidervenstrejustertuansettomikonellerikke.',
};
WithLongTextBreaking.argTypes = {
  ...WithLongTextBreaking.argTypes,
  children: { table: { disable: false } },
};

// Når Link har en veldig lang tekst, valgfritt ikon og eksternal ikon så skal tekst venstrejusteres
export const WithLongTextIconAndExternalIcon = Template.bind({});
WithLongTextIconAndExternalIcon.storyName =
  'With Long Text And Icons (A1 delvis)';
WithLongTextIconAndExternalIcon.args = {
  ...defaultArgs,
  isExternal: true,
  svgPath: CalendarSVGpath,
  children:
    'Denne lenken har en veldig lang tekst med ikon på venstre side. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er venstrejustert.',
};
WithLongTextIconAndExternalIcon.argTypes = {
  ...WithLongTextIconAndExternalIcon.argTypes,
  isExternal: { table: { disable: false } },
  svgPath: { table: { disable: false } },
  children: { table: { disable: false } },
};

// Når Link har ett valgfritt ikon, så vises ikonet. Tester også for de icon props som er relevant for saken med systemIcon som er brukt
export const WithIcon = Template.bind({});
WithIcon.storyName = 'With Icon (A4, B2)';
WithIcon.args = {
  ...defaultArgs,
  svgPath: CalendarSVGpath,
};
WithIcon.argTypes = {
  ...WithIcon.argTypes,
  svgPath: { table: { disable: false } },
};
WithIcon.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const link = canvas.getByRole('link');
  // eslint-disable-next-line testing-library/no-node-access
  const svg = link.querySelector('svg');
  await expect(svg).toHaveAttribute('aria-hidden', 'true');
  await expect(svg).toHaveAttribute('viewBox', systemIconViewBox);
};

// Når Link har isExternal, så vises riktig ikon med tilhørende tekst. Tester også for de icon props som er relevant for saken med systemIcon som er brukt
export const WithExternalIcon = Template.bind({});
WithExternalIcon.storyName = 'With External Icon (A5)';
WithExternalIcon.args = {
  ...defaultArgs,
  isExternal: true,
};
WithExternalIcon.argTypes = {
  ...WithExternalIcon.argTypes,
  isExternal: { table: { disable: false } },
};
WithExternalIcon.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const link = canvas.getByRole('link');
  // eslint-disable-next-line testing-library/no-node-access
  const svg = link.querySelector('svg');
  await expect(svg).toHaveAttribute('aria-label', 'Til et annet nettsted');
  await expect(svg).toHaveAttribute('viewBox', systemIconViewBox);
};

// Når Link har color white, så vises tekst og ikon i hvit
export const WithColor = Template.bind({});
WithColor.storyName = 'With Color (A6)';
WithColor.args = {
  ...defaultArgs,
  color: 'white',
  isExternal: true,
  svgPath: AddOutlineSVGpath,
};
WithColor.argTypes = {
  ...WithColor.argTypes,
  color: {
    options: ['default', ...linkColorArr],
    mapping: {
      default: '',
      ...linkColorArr,
    },
    control: 'inline-radio',
    table: {
      disable: false,
    },
  },
};
WithColor.parameters = {
  backgrounds: {
    default: 'themePrimary',
  },
  imageSnapshot: {
    hover: `${wrapper} > a`,
    focus: `${wrapper} > a`,
  },
};

// Når Link har target, så har link-elementet target med riktig verdi
export const WithTarget = Template.bind({});
WithTarget.storyName = 'With Target (A2)';
WithTarget.args = {
  ...defaultArgs,
  target: '_blank',
};
WithTarget.argTypes = {
  ...WithTarget.argTypes,
  target: { table: { disable: false } },
};
WithTarget.parameters = {
  imageSnapshot: { disable: true },
};

WithTarget.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const link = canvas.getByRole('link');
  expect(link).toHaveAttribute('rel', 'noreferrer');
  expect(link).toHaveAttribute('target', '_blank');
};

// Når Link har ariaDescribedby, så har link element aria-describedby satt
export const WithAriaDescribedby = Template.bind({});
WithAriaDescribedby.storyName = 'With AriaDescribedby (B3)';
WithAriaDescribedby.args = {
  ...defaultArgs,
  ariaDescribedby: elementId,
};
WithAriaDescribedby.argTypes = {
  ...WithAriaDescribedby.argTypes,
  ariaDescribedby: { table: { disable: false } },
};
WithAriaDescribedby.parameters = {
  imageSnapshot: { disable: true },
};
WithAriaDescribedby.play = verifyAttribute('aria-describedby', elementId);

// Når brukeren klikker på lenken, så kalles funksjonen i onClick prop.
// onClick-event endrer lenketeksten til lenken
const nyLinkText = 'Ny lenketekst slik at vi ser at event fungerte';
const OnClickTemplate: ComponentStory<typeof Link> = (args) => {
  const [linkText, setLinkText] = useState(
    'Klikk på lenken for å teste onClick event'
  );
  return (
    <div className={'noTransition'} data-test-block>
      <Link
        {...args}
        onClick={(e): void => {
          e.preventDefault();
          setLinkText(nyLinkText);
          args.onClick && args.onClick(e);
        }}
      >
        {linkText}
      </Link>
    </div>
  );
};
export const WithOnClick = OnClickTemplate.bind({});
WithOnClick.storyName = 'With onClick (A3 delvis)';
WithOnClick.args = {
  ...defaultArgs,
  svgPath: CalendarSVGpath,
};
WithOnClick.argTypes = {
  ...WithOnClick.argTypes,
  svgPath: { table: { disable: false } },
  onClick: { table: { disable: false } },
};
WithOnClick.parameters = {
  imageSnapshot: { disbale: true },
};
WithOnClick.play = async ({ args, canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const link = canvas.getByRole('link');
  await expect(link).toHaveTextContent(
    'Klikk på lenken for å teste onClick event'
  );
  await userEvent.click(link);
  await expect(link).toHaveTextContent(
    'Ny lenketekst slik at vi ser at event fungerte'
  );
  await waitFor(() => expect(args.onClick).toHaveBeenCalled());
};
