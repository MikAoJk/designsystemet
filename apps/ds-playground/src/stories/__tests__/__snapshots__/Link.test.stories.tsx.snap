// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Link Defaults smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     href="#root"
  >
    Er du pendler?
  </a>
</div>
`;

exports[`Tester/Link WithAriaDescribedby smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     href="#root"
     aria-describedby="htmlId"
  >
    Er du pendler?
  </a>
</div>
`;

exports[`Tester/Link WithColor smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY Link_link_white__bmeQu "
     href="#root"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo Link_icon__iy55p Link_icon_isCustom__nsOot"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8Zm0-18C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Zm1 5h-2v4H7v2h4v4h2v-4h4v-2h-4V7Z">
      </path>
    </svg>
    Er du pendler?
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo Link_icon__iy55p Link_icon_isExternal__O+HIA"
         viewbox="0 0 24 24"
         focusable="false"
         aria-label="shared.ExternalIcon"
         aria-hidden="false"
         role="img"
    >
      <path d="M14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7Zm5 16H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7h-2v7Z">
      </path>
    </svg>
  </a>
</div>
`;

exports[`Tester/Link WithCustomCss smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  dummyClassname"
     href="#root"
  >
    Er du pendler?
  </a>
</div>
`;

exports[`Tester/Link WithDataTestid smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     data-testid="123ID"
     href="#root"
  >
    Er du pendler?
  </a>
</div>
`;

exports[`Tester/Link WithExternalIcon smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     href="#root"
  >
    Er du pendler?
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo Link_icon__iy55p Link_icon_isExternal__O+HIA"
         viewbox="0 0 24 24"
         focusable="false"
         aria-label="shared.ExternalIcon"
         aria-hidden="false"
         role="img"
    >
      <path d="M14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7Zm5 16H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7h-2v7Z">
      </path>
    </svg>
  </a>
</div>
`;

exports[`Tester/Link WithIcon smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     href="#root"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo Link_icon__iy55p Link_icon_isCustom__nsOot"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M19 19H5V8h14v11ZM16 1v2H8V1H6v2H5c-1.11 0-2 .89-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-1V1h-2Zm1 11h-5v5h5v-5Z">
      </path>
    </svg>
    Er du pendler?
  </a>
</div>
`;

exports[`Tester/Link WithId smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a id="htmlId"
     class="Link_link__jeUpY  "
     href="#root"
  >
    Er du pendler?
  </a>
</div>
`;

exports[`Tester/Link WithLang smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     lang="nb"
     href="#root"
  >
    Er du pendler?
  </a>
</div>
`;

exports[`Tester/Link WithLongText smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     href="#root"
  >
    Denne lenken har en veldig lang tekst. Så lang at den lange teksten tvinger fram linjeskift med tekst som alltid er venstrejustert uansett om ikon eller ikke.
  </a>
</div>
`;

exports[`Tester/Link WithLongTextBreaking smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     href="#root"
  >
    Dennelenkenharenveldiglangtekst.Sålangatdentvingerframlinjeskiftmedtekstsomalltidervenstrejustertuansettomikonellerikke.
  </a>
</div>
`;

exports[`Tester/Link WithLongTextIconAndExternalIcon smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     href="#root"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo Link_icon__iy55p Link_icon_isCustom__nsOot"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M19 19H5V8h14v11ZM16 1v2H8V1H6v2H5c-1.11 0-2 .89-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-1V1h-2Zm1 11h-5v5h5v-5Z">
      </path>
    </svg>
    Denne lenken har en veldig lang tekst med ikon på venstre side. Så lang at den lange teksten tvinger fram linjeskift hvor tekst er venstrejustert.
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo Link_icon__iy55p Link_icon_isExternal__O+HIA"
         viewbox="0 0 24 24"
         focusable="false"
         aria-label="Til et annet nettsted"
         aria-hidden="false"
         role="img"
    >
      <path d="M14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7Zm5 16H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7h-2v7Z">
      </path>
    </svg>
  </a>
</div>
`;

exports[`Tester/Link WithOnClick smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     href="#root"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo Link_icon__iy55p Link_icon_isCustom__nsOot"
         viewbox="0 0 24 24"
         focusable="false"
         aria-hidden="true"
         role="img"
    >
      <path d="M19 19H5V8h14v11ZM16 1v2H8V1H6v2H5c-1.11 0-2 .89-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-1V1h-2Zm1 11h-5v5h5v-5Z">
      </path>
    </svg>
    Klikk på lenken for å teste onClick event
  </a>
</div>
`;

exports[`Tester/Link WithRef smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     href="#root"
     id="dummyIdForwardedFromRef"
  >
    Er du pendler?
  </a>
</div>
`;

exports[`Tester/Link WithTarget smoke-test 1`] = `
<div class="noTransition"
     data-test-block="true"
>
  <a class="Link_link__jeUpY  "
     href="#root"
     rel="noreferrer"
     target="_blank"
  >
    Er du pendler?
  </a>
</div>
`;
